internal/api/controllers/private/highlevelConnectionStatus.go

```diff
@@ -0,0 +1,240 @@
+package private
+
+import (
+	"fmt"
+	"net/http"
+	"playbook-dispatcher/internal/api/connectors"
+	"playbook-dispatcher/internal/api/connectors/inventory"
+	"playbook-dispatcher/internal/api/connectors/sources"
+	"playbook-dispatcher/internal/api/controllers/public"
+	"playbook-dispatcher/internal/common/utils"
+
+	"github.com/labstack/echo/v4"
+)
+
+type rhcSatellite struct {
+	SatelliteInstanceID      string
+	SatelliteOrgID           string
+	SatelliteVersion         string
+	Hosts                    []string
+	SourceID                 string
+	RhcClientID              *string
+	SourceAvailabilityStatus *string
+}
+
+func (this *controllers) ApiInternalHighlevelConnectionStatus(ctx echo.Context) error {
+	var input HostsWithOrgId
+	satelliteResponses := []RecipientWithConnectionInfo{}
+	directConnectedResponses := []RecipientWithConnectionInfo{}
+	noRHCResponses := []RecipientWithConnectionInfo{}
+
+	err := utils.ReadRequestBody(ctx, &input)
+	if err != nil {
+		utils.GetLogFromEcho(ctx).Error(err)
+		return ctx.NoContent(http.StatusBadRequest)
+	}
+
+	hostConnectorDetails, err := this.inventoryConnectorClient.GetHostConnectionDetails(
+		ctx.Request().Context(),
+		input.Hosts,
+		this.config.GetString("inventory.connector.ordered.how"),
+		this.config.GetString("inventory.connector.ordered.by"),
+		this.config.GetInt("inventory.connector.limit"),
+		this.config.GetInt("inventory.connector.offset"),
```

Comment:
I don't think the offset can be configurable.  Can it?

If the len(input.Hosts) > limit, then you would need to iterate through the pages.